lib/src/main/kotlin/org/walletconnect/impls/WCSession.kt

```diff
@@ -207,8 +211,12 @@ class WCSession(
             is Session.MethodCall.SessionUpdate -> {
                 if (!data.params.approved) {
                     endSession()
+                } else {
+                    updateSession(data.params)
+                    propagateToCallbacks {
+                        onStatus(Session.Status.Approved)
```

Comment:
TOL: would it be worth adding a new enum? `Session.Status.Updated`?

Reply:
Hm, yeah - that's probably clearer.

Reply:
Done